### PR TITLE
[thci] prefix fix for TH 1.1 border router 7.x.x cases

### PR DESCRIPTION
**Pull request by JaneFromSilabs**
Per https://threadgroup.atlassian.net/browse/DEV-2164

### CODE REVIEW

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: 

good to have a temporary fix before GRL delivers the final fix.

File: tools/harness-thci/OpenThread.py [hunk 1]
```diff
@@ -1805,15 +1805,16 @@ def configBorderRouter(
         if TESTHARNESS_VERSION == TESTHARNESS_1_2 and P_dp == 0 and not self.__isOpenThreadRunning():
             self.__useDefaultDomainPrefix = False
 
-        if TESTHARNESS_VERSION == TESTHARNESS_1_2:
+        if TESTHARNESS_VERSION == TESTHARNESS_1_2 and self.IsBorderRouter:
```
Comment (simonlingoogle) [inline tools/harness-thci/OpenThread.py:1808]:
```suggestion
        if TESTHARNESS_VERSION == TESTHARNESS_1_2 and self.IsBackboneRouter:
```

Every device using OpenThread_BR.py THCI will be `IsBorderRouter`, but a BR can also be DUT for 1.1 certification. So, we need to use 1.1 prefix rules for such BR device.

Comment (JaneFromSilabs) [reply tools/harness-thci/OpenThread.py]: Tested and working.  Thanks, Simon!

Comment (simonlingoogle) [reply tools/harness-thci/OpenThread.py]: `configBorderRouter` is defined in class `OpenThreadTHCI`. Isn't OpenThreadTHCI used by both OpenThread.py and OpenThread_BR.py ?

**Pull request merged**